drafts/draft-ietf-ippm-ioam-data-integrity.xml

```diff
@@ -912,18 +894,59 @@
       protection of IOAM-Data-Fields. The details of the key exchange are
       outside the scope of this document.</t>
 
-      <t>The Integrity Protection Method defined in this document requires
-      additional per-packet processing by each node that uses it. Inappropriate
-      use of the Integrity Protection Method might overload nodes and cause them
-      to stop functioning properly. Operators deploying IOAM with the Integrity
-      Protection Method MUST ensure that such overload situations are avoided.
-      This could for example be achieved by applying IOAM only to a subset of
-      the entire traffic.</t>
-
-      <t>The Nonce makes a signature chain unique but does not necessarily
-      prevent replay attacks. To enable replay protection, the encapsulating node
-      and the Validator(s) MUST agree on a common methodology to keep the Nonce valid
-      only for a specific period of time, which is outside the scope of this document.</t>
+      <t>There is an additional per-packet processing for each node that uses
+      the Integrity Protection Method defined in this document. Inappropriate
+      use of this Integrity Protection Method might overload nodes and cause
+      them to stop functioning properly. Operators deploying IOAM with this
+      Integrity Protection Method MUST ensure that such overload situations are
+      avoided. This could for example be achieved by applying IOAM only to a
+      subset of the entire traffic.</t>
+
+      <t>To ensure integrity protection of IOAM-Data-Fields, the Integrity
+      Protection Method defined in this document uses AES-GMAC
+      (<xref target="NIST.800-38D"/>). In that context, a generated key MUST be
+      fresh. Another important requirement is that the same combination of a
+      nonce (AES-GMAC IV) and a key MUST NOT be used more than once. Otherwise,
+      security guarantees are destroyed. To avoid such scenario, and to avoid
+      frequent rotation or refreshing of keys, it is RECOMMENDED to use a
+      12-octet (i.e., 96-bit) nonce based on the "Deterministic Construction"
+      (see <xref target="NIST.800-38D"/>, Sec. 8). It is defined as follows:
+        <list style="bullets">
+          <t>The nonce is the concatenation of two fields, called the fixed
+          field and the invocation field.</t>
+
+          <t>The fixed field identifies the device and MUST be unique. It can be
```

Comment:
Is there a potential to re-use the node_id?
Node_id short is 3 bytes (which could fit into the 32 bits needed here), Node_id wide is 7 bytes.

Reply:
Good catch, thanks! It was indeed the plan, so that the Validator can easily identify the encapsulating node and use the right key. I'll add it to the recommendations.

Reply:
Should we force the recommendations for the nonce construction too, i.e., s/RECOMMENDED/MUST/g ? Obviously for interop reasons, but no strong opinion though.

Reply:
MUST limits options - which is a good thing in this case

Reply:
> MUST limits options - which is a good thing in this case

Not sure what part you referred to here?

Reply:
I was referring to your statement; "Should we force the recommendations for the nonce construction too, i.e., s/RECOMMENDED/MUST/g ? Obviously for interop reasons, but no strong opinion though."

IMHO changing to MUST is reasonable.

Reply:
Got it! Sorry, I thought it was some proposed text :-D

Reply:
I suggest to s/RECOMMENDED/MUST/g and get feedback from the WG on this. WDYT?

Reply:
sounds good